src/authenticationservices.cs

```diff
@@ -0,0 +1,195 @@
+//
+// AuthenticationServices bindings
+//
+// Authors:
+//	Sebastien Pouliot  <sebastien.pouliot@microsoft.com>
+//
+// Copyright 2018 Microsoft Inc. All rights reserved.
+//
+
+using System;
+using Foundation;
+using ObjCRuntime;
+using UIKit;
+
+namespace AuthenticationServices {
+
+	[iOS (12,0)]
+	[Native]
+	[ErrorDomain ("ASCredentialIdentityStoreErrorDomain")]
+	public enum ASCredentialIdentityStoreErrorCode : long {
+		InternalError = 0,
+		StoreDisabled = 1,
+		StoreBusy = 2,
+	}
+
+	[iOS (12,0)]
+	[Native]
+	[ErrorDomain ("ASExtensionErrorDomain")]
+	public enum ASExtensionErrorCode : long {
+		Failed = 0,
+		UserCanceled = 1,
+		UserInteractionRequired = 100,
+		CredentialIdentityNotFound = 101,
+	}
+
+	[iOS (12,0)]
+	[Native]
+	public enum ASCredentialServiceIdentifierType : long {
+		Domain,
+		Url,
+	}
+
+	[iOS (12,0)]
+	[Native]
+	[ErrorDomain ("ASWebAuthenticationSessionErrorDomain")]
+	public enum ASWebAuthenticationSessionErrorCode : long {
+		CanceledLogin = 1,
+	}
+
+	[iOS (12,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface ASCredentialIdentityStore {
+		[Static]
+		[Export ("sharedStore")]
+		ASCredentialIdentityStore SharedStore { get; }
+
+		[Async]
+		[Export ("getCredentialIdentityStoreStateWithCompletion:")]
+		void GetCredentialIdentityStoreState (Action<ASCredentialIdentityStoreState> completion);
+
+		[Async]
+		[Export ("saveCredentialIdentities:completion:")]
+		void SaveCredentialIdentities (ASPasswordCredentialIdentity[] credentialIdentities, [NullAllowed] Action<bool, NSError> completion);
+
+		[Async]
+		[Export ("removeCredentialIdentities:completion:")]
+		void RemoveCredentialIdentities (ASPasswordCredentialIdentity[] credentialIdentities, [NullAllowed] Action<bool, NSError> completion);
+
+		[Async]
+		[Export ("removeAllCredentialIdentitiesWithCompletion:")]
+		void RemoveAllCredentialIdentities ([NullAllowed] Action<bool, NSError> completion);
+
+		[Async]
+		[Export ("replaceCredentialIdentitiesWithIdentities:completion:")]
+		void ReplaceCredentialIdentities (ASPasswordCredentialIdentity[] newCredentialIdentities, [NullAllowed] Action<bool, NSError> completion);
+	}
+
+	[iOS (12,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface ASCredentialIdentityStoreState {
+		[Export ("enabled")]
+		bool Enabled { [Bind ("isEnabled")] get; }
+
+		[Export ("supportsIncrementalUpdates")]
+		bool SupportsIncrementalUpdates { get; }
+	}
+
+	[iOS (12,0)]
+	[BaseType (typeof (NSExtensionContext))]
+	[DisableDefaultCtor]
+	interface ASCredentialProviderExtensionContext {
+		[Export ("completeRequestWithSelectedCredential:completionHandler:")]
+		void CompleteRequest (ASPasswordCredential credential, [NullAllowed] Action<bool> completionHandler);
```

Comment:
Same, I would use a named delegate here for `Action<bool>`, what does that `bool` means?